docs: add "Improved Patch Delivery" blog (#158)
shorebirdtech/website: src/pages/blog/improved-patch-delivery.md

---
layout: ../../layouts/BlogLayout.astro
title: Blog | Improved Patch Delivery
description: Announcing improved global patch delivery, availability, and performance.
date: 9/5/2024
---

![Improved patch delivery header image](../../assets/images/blog/improved-patch-delivery/heading.png)

> We're excited to announce some improvements to our cloud infrastructure which
> improve our patch delivery, availability, and performance around the world 🥳

Reviewer: ```suggestion
We're excited to announce some improvements to cloud infrastructure which
improve patch delivery, availability, and performance around the world 🥳
```

## What's Changed

> When a patch is published via `shorebird patch`, the generated patch artifact is uploaded to Shorebird's cloud storage.

Reviewer: ```suggestion
When a patch is published via `shorebird patch`, the generated patch artifact is uploaded to Shorebird's cloud.
```

On app launch, the corresponding release running on users' devices will ping Shorebird's API to check whether a newer patch is available.
If a newer patch is available, Shorebird's API will respond with a link to the patch artifact and the patch is then downloaded and installed on the device.

> We've made some improvements to how we store and serve patch artifacts to both improve availability (e.g patches can be downloaded by devices in more places around the world) and performance (e.g patches take less time to download).

Reviewer 2: ```suggestion
We've made some improvements to how we store and serve patch artifacts to both improve availability and performance—patches can be downloaded faster in more places around the world.
```

> ## Wider Availability Worldwide

Reviewer 2: ```suggestion
## Wider Availability
```

We've made some changes to our cloud infrastructure to enable Shorebird to deliver patches to more regions around the world. You may now find that users are able to download patches in regions that were previously unavailable.

![China Patch availability image](../../assets/images/blog/improved-patch-delivery/china-availability.png)

> **Note: There may still be regions in which Shorebird is unavailable. Please [contact us](mailto:contact@shorebird.dev) if you notice users having trouble downloading patches and we're happy to help.**

Reviewer: ```suggestion
```

Reviewer: I don't think this adds much.

> ## Better Performance Worldwide

Reviewer 2: ```suggestion
## Faster Downloads
```
